spark/src/main/scala/org/apache/spark/sql/delta/GenerateIdentityValues.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import com.databricks.spark.util.MetricDefinitions
+import com.databricks.spark.util.TagDefinitions.TAG_OP_TYPE
+import org.apache.spark.sql.delta.metering.DeltaLogging
+
+import org.apache.spark.{SparkException, TaskContext}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Expression, LeafExpression, Nondeterministic}
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, CodeGenerator, ExprCode, FalseLiteral}
+import org.apache.spark.sql.catalyst.expressions.codegen.Block._
+import org.apache.spark.sql.types.{DataType, LongType}
+
+/**
+ * Returns the next generated IDENTITY column value based on the underlying
+ * [[PartitionIdentityValueGenerator]].
+ */
+case class GenerateIdentityValues(
+    generator: PartitionIdentityValueGenerator) extends LeafExpression with Nondeterministic {
```

Comment:
Could we put the constructor on the same line and the extends on a separate line? I think it's more consistent with the rest of the codebase. :)